public/controllers/agent/components/register-agent.js

```diff
@@ -118,25 +173,27 @@ export class RegisterAgent extends Component {
   }
 
   selectOS(os) {
-    this.setState({ selectedOS: os });
+    this.setState({ selectedOS: os, selectedVersion: '', selectedArchitecture: '' });
   }
 
   setServerAddress(event) {
     this.setState({ serverAddress: event.target.value });
   }
 
-  setWazuhPassword(event) {
-    this.setState({ wazuhPassword: event.target.value });
+  setGroupName(selectedGroup) {
+    this.setState({ selectedGroup });
   }
 
-  /**
-   * Checks if the password is not needed, in that case remove the input password step
-   * @param {Array} steps
-   */
-  cleanSteps(steps) {
-    if (!this.state.needsPassword || this.state.hidePasswordInput)
-      steps.splice(2, 1);
-    return steps;
+  setArchitecture(selectedArchitecture) {
+    this.setState({ selectedArchitecture });
+  }
+
+  setVersion(selectedVersion) {
+    this.setState({ selectedVersion });
+  }
```

Comment:
Should we restart the `selectedArchitecture` variable (set to empty strin '') when a `selectedVersion` (OS version) is selected?